### ISSUE TITLE
typelevel dependency

### ISSUE THREAD

**Issue by jamesdbrock** [purescript-contrib/purescript-arraybuffer#29]
For moving this package to __purescript-contrib__, @thomashoneyman has requested that the package has no dependencies outside of __purescript-contrib__, and the __purescript-typelevel__ package does not belong to __purescript-contrib__.

The `BytesPerValue` class with the dependency on __purescript-typelevel__ was introduced in this redesign by @athanclark .

https://github.com/jacereda/purescript-arraybuffer/pull/23

Do you have any advice about this @jacereda @athanclark ? How would you approach this?

**Comment by jamesdbrock** [comment]
From __purescript-typelevel__ we only use the `toInt'` method of the `Nat` class on types `D1`, `D2`, `D4`, and `D8`.

Here's one thing which I think we could do which I think would completely eliminate the need for __purescript-typelevel__.

We could replace the `BytesPerValue` typeclass with something like this:

```haskell
class BytesPerType (a :: ArrayViewType) where
  byteSize :: (Proxy a) -> Int

instance bytesPerTypeInt8 :: BytesPerType Int8 where byteSize _ = 1
instance bytesPerTypeInt16 :: BytesPerType Int16 where byteSize _ = 2
…
```

Then we would also be able to remove the `Nat` type constraint on a lot of public API functions, which would be a win in my opinion.

**Comment by jamesdbrock** [comment]
Resolved in v11.0.0 .
